goal/src/main/java/org/openapitools/vertxweb/server/api/impl/PetsApiImpl.java

```diff
@@ -24,20 +26,23 @@ public PetsApiImpl(RouterBuilder builder) {
     }
 
     @Override
-    public PetsApi createPetsHandler(Handler<RoutingContext> handler) {
-        builder.operation("createPets").handler(handler);
+    public PetsApi createPetsHandler(CreatePetsHandler handler) {
+        builder.operation("createPets")
+                .handler(ctx -> handler.handle(ctx));
         return this;
     }
 
     @Override
-    public PetsApi listPetsHandler(Handler<RoutingContext> handler) {
-        builder.operation("listPets").handler(handler);
+    public PetsApi listPets(ListPetsHandler handler) {
+        builder.operation("showPetById")
+                .handler(ctx -> handler.handle(ctx, ctx.get("limit")));
         return this;
     }
 
     @Override
-    public PetsApi showPetsByIdHandler(Handler<RoutingContext> handler) {
-        builder.operation("showPetById").handler(handler);
+    public PetsApi showPetByIdHandler(ShowPetByIdHandler handler) {
+        builder.operation("showPetById")
+                .handler(ctx -> handler.handle(ctx, ctx.get("petId")));
```

Comment:
Why this "dance" of putting the param inside the context? What if you just wrap the `ShowPetByIdHandler` in the `preShowPetById` logic?

Reply:
The reason is that as we can have multiple handlers on the same operation, think, a chain of small actions, authz, logging, data access, transform, return. This ensures that the parsing only happens one time. No need to do it multiple times.

The cost is minimal but we could make it a template fragment and parse every time. 

Right now it's just an implementation detail (not really affecting the public facing api)

Reply:
> This ensures that the parsing only happens one time. No need to do it multiple times.

`RequestParameters` is not parsing anything, the parsing is done in the `ValidationHandler`. `RequestParameters` is just the container of the results.